src/components/NavBar.js

```diff
@@ -1,33 +1,45 @@
-import React, { Component } from 'react';
-import { NavLink } from 'react-router-dom';
-import { Container, Menu, Input } from 'semantic-ui-react';
-import '../styling/Navbar.css';
-import AlertMessage from './AlertMessage';
-import { connect } from 'react-redux';
+import React, { Component } from "react";
+import { NavLink } from "react-router-dom";
+import { Container, Menu, Input } from "semantic-ui-react";
+import "../styling/Navbar.css";
+import AlertMessage from "./AlertMessage";
+import { connect } from "react-redux";
+import UpdateLanguage from './UpdateLanguage';
+import i18n from "../i18n";
 
 class NavBar extends Component {
-  state = { activeItem: 'latest news' }
+  constructor(props) {
+    super(props)
+    this.ReRender = this.ReRender.bind(this);
+  };
+  handleItemClick = (e, { name }) => this.setState({ activeItem: name });
 
-  handleItemClick = (e, { name }) => this.setState({ activeItem: name })
+  ReRender() {
+    this.forceUpdate();
+  }
 
   render() {
     let createArticleButton;
     let subscribeButton;
     let flashMessage;
     let loginActions;
 
-    const { activeItem } = this.state
+    const { t } = this.props;
 
     if (this.props.showFlash === true) {
-      flashMessage = <AlertMessage />
+      flashMessage = <AlertMessage />;
```

Comment:
Is this some legacy code? What is it used for? It seems that your branch is not up to date with `development`

Reply:
This `AlertMessage` component is in the up to date iteration of the `NavBar` component in `development`.  It is linked to the AlertMessage.js file which shows, in part, the red and green - failure and success borders to flash messages.